src/Benchmarks.tsx

```diff
@@ -21,87 +28,173 @@ interface Props {
   columns: Column[];
   yLabel?: string;
   sha1List: string[];
+  scale: "linear" | "log";
 }
 
+const COLORS = [
+  "#a6cee3",
+  "#1f78b4",
+  "#b2df8a",
+  "#33a02c",
+  "#fb9a99",
+  "#e31a1c",
+  "#fdbf6f",
+  "#ff7f00",
+  "#cab2d6",
+  "#6a3d9a",
+  "#ffff99",
+  "#b15928"
+];
+
 function BenchmarkChart(props: Props) {
   const theme = useTheme<Theme>();
-
-  function viewCommitOnClick(c1: any, c2: any, { dataPointIndex }: any): void {
-    window.open(
-      `https://github.com/denoland/deno/commit/${props.sha1List[dataPointIndex]}`
-    );
-  }
-
-  if (props.yTickFormat && props.yTickFormat === formatLogScale) {
-    logScale(props.columns);
+  const [hovered, setHovered] = useState<string>(null);
+  const [disabled, setDisabled] = useState<string[]>([]);
+  const [refLeft, setRefLeft] = useState<number>(null);
+  const [refRight, setRefRight] = useState<number>(null);
+  const [boundLeft, setBoundLeft] = useState<number>(null);
+  const [boundRight, setBoundRight] = useState<number>(null);
+
+  function viewCommitOnClick(d: number): void {
+    window.open(`https://github.com/denoland/deno/commit/${sha1List[d]}`);
   }
 
-  const options = {
-    theme: {
-      mode: theme.palette.type,
-      palette: "palette1"
-    },
-    colors: [
-      "#a6cee3",
-      "#1f78b4",
-      "#b2df8a",
-      "#33a02c",
-      "#fb9a99",
-      "#e31a1c",
-      "#fdbf6f",
-      "#ff7f00",
-      "#cab2d6",
-      "#6a3d9a",
-      "#ffff99",
-      "#b15928"
-    ],
-    chart: {
-      background: theme.palette.background.default,
-      foreColor: theme.palette.getContrastText(
-        theme.palette.background.default
-      ),
-      toolbar: {
-        show: true
-      },
-      animations: {
-        enabled: false
-      },
-      events: {
-        markerClick: viewCommitOnClick
-      }
-    },
-    stroke: {
-      width: 2,
-      curve: "straight"
-    },
-    legend: {
-      show: true,
-      showForSingleSeries: true,
-      position: "bottom"
-    },
-    yaxis: {
-      labels: {
-        formatter: props.yTickFormat
-      },
-      title: {
-        text: props.yLabel
-      }
-    },
-    xaxis: {
-      labels: {
-        show: false
-      },
-      categories: props.sha1List,
-      tooltip: {
-        enabled: false
+  function zoom() {
```

Comment:
Isn't there a library that implements zoom for us? I don't want to maintain UI code.